### ISSUE TITLE
[BUG] minVersion is failing on ranges like `^0 >=0.1.2`

### ISSUE THREAD

**Issue by SkReD** [npm/node-semver#330]
# What / Why
`.minVersion` is failing with error on range `^0 >=0.1.2`. Workaround is`minVersion(validRange('<range>'))`

### Current Behavior
Failing with `undefined` error

### Expected Behavior
Returns `0.1.2`

## References
`semver` version - `7.3.2`

**Comment by dalyr95** [comment]
Got the same issue in the above Issue with npm/cli and it's engines. minVersion returns null

```
const semver = require('semver');
undefined
> semver.minVersion("6 >=6.2.0 || 8 || >=9.3.0");
null
> semver.minVersion(">=6.2.0 || 8 || >=9.3.0");
SemVer {
  options: { loose: false, includePrerelease: false },
  loose: false,
  includePrerelease: false,
  raw: '6.2.0',
  major: 6,
  minor: 2,
  patch: 0,
  prerelease: [],
  build: [],
  version: '6.2.0'
}
> semver.minVersion("6 || >=6.2.0 || 8 || >=9.3.0");
SemVer {
  options: { loose: false, includePrerelease: false },
  loose: false,
  includePrerelease: false,
  raw: '6.0.0',
  major: 6,
  minor: 0,
  patch: 0,
  prerelease: [],
  build: [],
  version: '6.0.0'
}
```

**Comment by lukekarrys** [comment]
Fixed by #341
